Early SMF initialization

Rupurudu opened the pull request:
Moved the SMF initialization from template class to the skin class. This allows us to use SMF's globals early. An example usage is added (Adds forum's favicon to the wiki).

This also fixes:
* Incorrect viewport setting. (copied from the curve theme)
* Missing theme color.

jdarwood007 commented:
An issue I see so far is that you changed the AJAX loading to use SMF's theme URL, which is only present when loaded with SSI.  If SSI is off, that won't set.

Rupurudu commented:
> An issue I see so far is that you changed the AJAX loading to use SMF's theme URL, which is only present when loaded with SSI. If SSI is off, that won't set.

This is kind of the point. The `reqOverlayDiv` function is essentially a copy-paste from SMF. While I could easily fix the function by including the loading gif into the mediawiki theme, I chose to load SMF's globals. This way we can use more of the SMF's scripts as-is, without modifications, which makes the development easier.

jdarwood007 commented:
> This is kind of the point. The `reqOverlayDiv` function is essentially a copy-paste from SMF. While I could easily fix the function by including the loading gif into the mediawiki theme, I chose to load SMF's globals. This way we can use more of the SMF's scripts as-is, without modifications, which makes the development easier.

This is great in theory, but this theme is designed to operate under the assumption that SMF is not installed, i.e., it is standalone.  If we wanted SMF fully, after loading SSI, we could have essentially called template_header/template_footer() and done some adjustments to capture the output for the logic that adds the scripts/css to inject them into SMF.

jdarwood007 commented:
Creating the integration version would be acceptable.  The idea for custom is to prepare some adjustments that allow for tweaking the provided template with changes.  We use a modified version of the custom on our site to overload the changes yet again to integrate deeply into our site usage.  But its otherwise the same as custom.

Rupurudu commented:
> Creating the integration version would be acceptable. The idea for custom is to prepare some adjustments that allow for tweaking the provided template with changes. We use a modified version of the custom on our site to overload the changes yet again to integrate deeply into our site usage. But its otherwise the same as custom.

Sorry for the late reply. I completely forgot about this.

About the installation on [wiki.simplemachines.org](https://wiki.simplemachines.org/), I have found a number of differences unrelated to the customizations, even on the core theme.

For example if we compare the default curve2 theme on the simplemachines site to a fresh installation from this repo you might find that the quick search is missing:
<img width="1212" height="353" src="https://github.com/user-attachments/assets/e17ce2b4-3e5b-4632-99bf-3048e729ee54" />

(You can view the uncustomized version here: [https://wiki.simplemachines.org/w/index.php?title=Main_Page&useskin=smfcurve2](https://wiki.simplemachines.org/w/index.php?title=Main_Page&useskin=smfcurve2))

This is something I fixed myself (yet to be upstreamed).

There is also the case of icons, where this repo has missing most of the icons while it's visible on simplemachines website:
<img width="650" height="84" src="https://github.com/user-attachments/assets/f08a06da-20ff-48aa-8266-467368ec99bf" />

Again, this is something I fixed.

Also the Mobile menu fix #31 I upstreamed here is not yet installed to simplemachines.

So, my question is: How extensive are these modifications?

I'm asking this because looking at this repo's activity, and the number of fixes present on simplemachines site and not this repo, this theme is only used on simplemachines site.

Honestly, I don't want to waste my time upstreaming, if it ends up being not used.

Thanks for understanding. I've spent a lot of time making many changes and adding many new useful features like, forum news, forum logo (instead of showing mediawiki logo), forum slogan, collapseable sidebar and a completely redesigned usermenu that mimics SMF's usermenu for standalone and merged SMF/MediaWiki usermenu for use with [smf-mw-auth](https://github.com/SimpleMachines/smf-mw-auth). And it'll take me time and effort to beautify up my code, seperate them up to chunks, and open merge requests.
